pages/profile/[username].tsx

```diff
@@ -31,6 +31,22 @@ export default function Profile(props: PageProps) {
         }
     }, [data])
 
+    const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+    useEffect(() => {
+        adjustTextareaHeight();
+    }, [message]);
+
+    const adjustTextareaHeight = () => {
+        if (textAreaRef.current) {
+            const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
+            textAreaRef.current.style.height = `${lineHeight}px`;
+            const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
+            const numLines = lines <= 3 ? lines : 3;
+            textAreaRef.current.style.height = `${numLines * lineHeight}px`;
+        }
+    };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new `useRef` hook and an `adjustTextareaHeight` function have been introduced. The `useRef` hook creates a reference to a textarea element, and the `adjustTextareaHeight` function dynamically adjusts the height of the textarea based on its content. This is a good practice for improving user experience by ensuring that the textarea always displays all its content without unnecessary scrolling.
```javascript
const textAreaRef = useRef<HTMLTextAreaElement>(null)

useEffect(() => {
    adjustTextareaHeight();
}, [message]);

const adjustTextareaHeight = () => {
    if (textAreaRef.current) {
        const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
        textAreaRef.current.style.height = `${lineHeight}px`;
        const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
        const numLines = lines <= 3 ? lines : 3;
        textAreaRef.current.style.height = `${numLines * lineHeight}px`;
    }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->